config.w32

```diff
@@ -140,6 +151,14 @@ if (PHP_MONGODB != "no") {
     WARNING("mongodb libsasl support not enabled, libs not found");
   }
 
+  if (typeof COMPILER_NAME === 'string') {
+    mongoc_opts.MONGOC_CC = COMPILER_NAME;
+  } else if (typeof VC_VERSIONS[VCVERS] === 'string') {
+    mongoc_opts.MONGOC_CC = VC_VERSIONS[VCVERS];
+  }
+  mongoc_opts.MONGOC_USER_SET_CFLAGS = get_define("CFLAGS_MONGODB");
+  mongoc_opts.MONGOC_USER_SET_LDFLAGS = get_define("LDFLAGS_MONGODB");
```

Comment:
This will result in something similar to:

```
/* MONGOC_USER_SET_CFLAGS is set from config based on what compiler flags were
 * used to compile mongoc */
#define MONGOC_USER_SET_CFLAGS "/D BSON_COMPILATION /D MONGOC_COMPILATION /D MONGOC_TRACE /I..\pecl\mongodb /I..\pecl\mongodb/src/libmongoc/src/mongoc /I..\pecl\mongodb/src/libbson/src/bson/ /I..\pecl\mongodb/src/libbson/src /I..\pecl\mongodb/src/contrib /I \"C:\php-sdk\phpdev\vc14\x86\deps\include\" /DHAVE_OPENSSL_SSL_H=1"

#define MONGOC_USER_SET_LDFLAGS "/libpath:\"..\deps\lib\" /libpath:\"..\deps\lib\""

/* MONGOC_CC is used to determine what C compiler was used to compile mongoc */
#define MONGOC_CC "MSVC14 (Visual C++ 2015)"
```

The `MONGOC_CC` constant is certainly useful, but I don't think CFLAGS and LDFLAGS are helpful. In libmongoc's `configure.ac`, the CFLAGS and LDFLAGS are set before the library's own configuration. Do we even expect users to have custom CFLAGS for Windows builds?

Also, the `LIBS_MONGODB` environment variable may be more useful, as it actually lists the linked libraries. `LDFLAGS_MONGODB` is only reporting the paths.